### ISSUE TITLE
'tensorflow.python.framework.ops.EagerTensor' object has no attribute 'mean'

### ISSUE THREAD

**Issue by jackaraz** [shap/shap#1164]
Hi, guys hope everyone is safe and healthy! I'm trying to get SHAP values from a Keras model where after loading the model and weights, I import the CSV data using pandas then got a slice as follows;

`background = df_full.iloc[np.array(df_full.index.array)[np.random.choice(df_full.shape[0], 10, replace=False)]]`
`background = background.reset_index(drop=True)`
`explainer = shap.DeepExplainer(model,tf.constant(background.values, dtype=np.float32))`
`X  = df_full.iloc[np.array(df_full.index.array)[np.random.choice(df_full.shape[0], 
                                                                 10, replace=False)]]`
`shap_values = explainer.shap_values(tf.constant(X.values, dtype=np.float32))`

But I get an error from `shap/explainers/deep/deep_tf.py` L308 saying as follows. 

`'tensorflow.python.framework.ops.EagerTensor' object has no attribute 'mean'`

In `deep_tf.py`, I found the corresponding line which was;

`phis[l][j] = (sample_phis[l][bg_data[l].shape[0]:] * (X[l][j] - bg_data[l])).mean(0)`

where this `.mean(0)` was problematic so I changed it with the following;

`phis[l][j] = np.mean(sample_phis[l][bg_data[l].shape[0]:] * (X[l][j] - bg_data[l]), axis=0)`

Now it's working for me but I wanted to ask if I'm doing any mistake here, looks like fine and results are as expected. Would be appreciated if you can shine some light on the matter. 

Cheers

**Comment by vfro** [comment]
I also bumped into the same issue and the fix proposed by @jackaraz helped. 

Please, include the fix into the next version.

Name: shap                                                                                                                                                                                                           Version: 0.35.0

Name: tensorflow                                                                                                                                                                                                     Version: 2.2.0

**Comment by liusycn** [comment]
I changed shap/explainers/deep/deep_tf.py L308 as @jackaraz suggested but the same error occurs. Any help is appreciated！

**Comment by bmwise14** [comment]
hello, will a fix be made to this problem in the latest updates of shap? I am using shap==0.37.0 with tensorflow==2.3.1. The same error mentioned at the beginning of the thread is still occurring.

**Comment by detrin** [comment]
Do any of you still have this issue? If you can please try it with the newest version of shape and tensorflow.

**Comment by Alexwong-1024** [comment]
I also encountered this problem, have you solved it? @jackaraz @detrin

**Comment by TarcisioBalbi** [comment]
I had the same issue. Turns out that shap expects the input data to be an np.array, not a tensor.
So, in case anyone runs into the same problem, just make sure your data structure is correct (same as your model expects), but instead of using tensors, convert your data to simple numpy arrays.

I'm using:
- shap: '0.46.0'
- tensorflow: '2.18.0'
